### PR TITLE
frontend changes for API response work

### PR DESCRIPTION
**Pull request by hamilton**
addresses frontend changes needed for #479. Instructions for pulling in / building the data are in that issue.

### CODE REVIEW

File: src/config/firefox-desktop.js [hunk 1]
```diff
@@ -48,6 +48,9 @@ export default {
     if (pr.record_in_processes[0] === 'all') {
       pr.record_in_processes = ['main', 'content', 'gpu'];
     }
+    if (pr.record_in_processes[0] === 'all_childs') {
+      pr.record_in_processes = ['main', 'content'];
```
Comment (robhudson) [inline src/config/firefox-desktop.js]: This should be `['content', 'gpu']`. All childs is everything but main (aka parent).

File: src/state/store.js [hunk 2]
```diff
@@ -263,15 +251,6 @@ export function extractBucketMetadata(transformedData) {
...
-export function isCategorical(probeType, probeKind) {
-  return (
-    (probeType === 'histogram' && probeKind === 'enumerated') ||
-    probeKind === 'categorical' ||
-    probeKind === 'flag' ||
-    probeKind === 'boolean'
-  );
-}
-
 export function fetchDataForGLAM(params) {
```
Comment (hamilton) [inline src/state/store.js:254]: @spasovski @openjck my hope is that w/ this PR, y'all can more easily follow along w/ changes to `store.js` and `probe-utils.js`. Let me know how I can help clarify anything in those two files.

Comment (spasovski) [reply src/state/store.js:254]: I see a lot of red so that helps

File: src/components/home/Quantile.svelte [hunk 3]
```diff
@@ -21,11 +19,14 @@ let width;
...
+    dist = probe.filter((d) => d.client_agg_type === 'summed-histogram');
+    dist = dist[dist.length - 1];
   } else if (metricType === 'scalar' && 'avg' in probe) {
-    dist = probe.data.find((d) => d.client_agg_type === 'avg');
+    dist = probe.filter((d) => d.client_agg_type === 'avg');
+    dist = dist[dist.length - 1];
   } else {
-    dist = probe.data[0];
+    dist = probe[probe.length - 1];
+    // dist = probe.histogram;
```
Comment (openjck) [inline src/components/home/Quantile.svelte]: Can we remove this or is it kept here for reference?

File: src/components/home/Quantile.svelte [hunk 4]
```diff
@@ -92,9 +93,8 @@ onMount(() => {
           <text in:fade={{ duration: 100 }} fill=var(--cool-gray-600) x={left} text-anchor=start  y={top - 4}>{formatCount(hoverValue.x)}</text>
           <text in:fade={{ duration: 100 }} fill=var(--cool-gray-600) x={right} text-anchor=end  y={top - 4}>{formatCount(hist[hoverValue.x])} clients</text>
         {/if}
-        <!-- <text  fill=var(--cool-gray-600) text-anchor=end x={right}  y={top - 4}>{formatCount($totalClients)} clients </text> -->
```
Comment (openjck) [inline src/components/home/Quantile.svelte:95]: Can we remove this or is it kept here for reference?

File: src/utils/probe-utils.js [hunk 5]
```diff
@@ -55,19 +45,20 @@ function createNewError(which) {
...
     }
     const histogram = responseHistogramToGraphicFormat(h);
-    const { percentiles } = probe.data[0];
+    const { percentiles } = probe;
 
     if (!percentiles) {
       throw createNewError('MISSING_PERCENTILES');
     }
 
+    // fixme: remove need for transformedPercentiles.
```
Comment (openjck) [inline src/utils/probe-utils.js]: Can we change this to `FIXME`? I tend to assume capitalization when searching for these keywords.

File: src/utils/probe-utils.js [hunk 6]
```diff
@@ -101,13 +90,26 @@ function toProportions(obj) {
...
+//   const topK = Object.values(byBuildID).map((matches) => {
+//     const out = matches;
+//     out.sort(sortByKey('audienceSize'));
+//     out.reverse();
+//     return out.slice(0, k);
+//   });
+//   const flattened = topK.flat(2);
+//   flattened.sort(sortByKey('label'));
+//   return flattened;
+// }
```
Comment (openjck) [inline src/utils/probe-utils.js]: Can we remove this or is it kept here for reference?

File: src/utils/probe-utils.js [hunk 7]
```diff
@@ -12,14 +10,6 @@ export function sortByKey(key) {
   };
 }
```
Comment (openjck) [inline src/utils/probe-utils.js:11]: I know this isn't part of this pull request, so I'm happy to open a follow-up issue about this, but in reading through this file, I noticed something. Should line 7 be `if (a < b)`? If they're equal we should return zero.

File: src/utils/probe-utils.js [hunk 8]
```diff
@@ -12,14 +10,6 @@ export function sortByKey(key) {
...
 
-export const sortByHistogramObjectKey = (a, b) => {
-  const ai = +a;
-  const bi = +b;
-  if (ai < bi) return -1;
-  if (ai >= bi) return 1;
-  return 0;
-};
-
 export const responseHistogramToGraphicFormat = (
```
Comment (openjck) [inline src/utils/probe-utils.js:13]: Also not part of this PR, really, but do we need to export this? I don't see it being used externally.

File: src/utils/probe-utils.js [hunk 9]
```diff
@@ -117,116 +119,33 @@ export const prepareForProportionPlot = (
...
-        metadata: { ...aggregation.metadata },
-      });
-    });
-  });
-  return gathered;
-}
-
-export function byKeyAndAggregation(
+export function transformResponse(
   d,
```
Comment (openjck) [inline src/utils/probe-utils.js:119]: I know this isn't part of the pull request, but since we're here, can we give this a more descriptive name?

**Review by openjck — COMMENTED** [review]
This all looks fine to me and works well.
